### PR TITLE
Whisper chat

### PR DESCRIPTION
**Pull request by BloodWod-513**

### CODE REVIEW

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -4478,3 +4496,30 @@ bool CGameContext::IsSpectatorCID(int ClientID) {
 	auto& vec = Server()->spectators_id;
 	return std::find(vec.begin(), vec.end(), ClientID) != vec.end();
 }
+
+void CGameContext::Converse(int ClientID, char *pStr)
```
Comment (bretonium) [inline src/game/server/gamecontext.cpp:4500]: We have a player-used command `/mute`. It was added in c6a5028 and i have later added it to /msg (5c603c6). `/c` should honor it

File: src/game/server/gamecontext.cpp [hunk 2]
```diff
@@ -4478,3 +4496,36 @@ bool CGameContext::IsSpectatorCID(int ClientID) {
...
+	if (!pPlayer)
+		return;
+	if(Server()->GetClientSession(ClientID) && Server()->GetClientSession(ClientID)->m_MuteTick > 0)
+	{
+		int Time = Server()->GetClientSession(ClientID)->m_MuteTick/Server()->TickSpeed();
+		SendChatTarget_Localization(ClientID, CHATCATEGORY_ACCUSATION, _("You are muted for {sec:Duration}"), "Duration", &Time, NULL);
+		return;
+	}
+
+	for(int i = 0; i < MAX_CLIENTS; i++)
```
Comment (bretonium) [inline src/game/server/gamecontext.cpp:4512]: i don't understand why the loop is required here and why `m_apPlayers[i]` is checked. Could you please explain? Maybe you could add a comment here that would explain what is going on here?

Comment (BloodWod) [reply src/game/server/gamecontext.cpp:4512]: Oh, this is my inattention, and really strange, I'll fix it now

**Comment by bretonium** [comment]
Sorry, but now i am getting this:
```
src/game/server/gamecontext.cpp: In member function ‘virtual void CGameContext::OnMessage(int, CUnpacker*, int)’:
src/game/server/gamecontext.cpp:1780:41: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
     Converse(ClientID, pMsg->m_pMessage + 10);
                        ~~~~~~~~~~~~~~~~~^~~~
In file included from src/game/server/gamecontext.cpp:11:0:
src/game/server/gamecontext.h:266:7: note:   initializing argument 2 of ‘void CGameContext::Converse(int, char*)’
  void Converse(int ClientID, char *pStr);
       ^~~~~~~~
src/game/server/gamecontext.cpp:1784:41: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
     Converse(ClientID, pMsg->m_pMessage + 3);
                        ~~~~~~~~~~~~~~~~~^~~
In file included from src/game/server/gamecontext.cpp:11:0:
src/game/server/gamecontext.h:266:7: note:   initializing argument 2 of ‘void CGameContext::Converse(int, char*)’
  void Converse(int ClientID, char *pStr);
       ^~~~~~~~
src/game/server/gamecontext.cpp: In member function ‘void CGameContext::Converse(int, char*)’:
src/game/server/gamecontext.cpp:4523:47: error: ‘Buffer’ was not declared in this scope
   TextIter = FinalMessage.append_at(TextIter, Buffer.buffer());
                                               ^~~~~~
```

File: src/game/server/gamecontext.cpp [hunk 3]
```diff
@@ -4526,4 +4526,4 @@ void CGameContext::Converse(int ClientID, const char* pStr)
 		Msg.m_pMessage = FinalMessage.buffer();
 		Server()->SendPackMsg(&Msg, MSGFLAG_VITAL, pPlayer->m_LastWhisperTo);
 	}
-}
```
Comment (bretonium) [inline src/game/server/gamecontext.cpp]: i don't understand what was fixed here :-) it seems that the compilation error that i posted was not fixed at all

Comment (bretonium) [reply src/game/server/gamecontext.cpp]: ok, the error is a little different
```
[52/77] #1 c++ src/game/server/gamecontext.cpp
src/game/server/gamecontext.cpp: In member function ‘virtual void CGameContext::OnMessage(int, CUnpacker*, int)’:
src/game/server/gamecontext.cpp:1791:41: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
     Converse(ClientID, pMsg->m_pMessage + 10);
                        ~~~~~~~~~~~~~~~~~^~~~
In file included from src/game/server/gamecontext.cpp:11:0:
src/game/server/gamecontext.h:267:7: note:   initializing argument 2 of ‘void CGameContext::Converse(int, char*)’
  void Converse(int ClientID, char *pStr);
       ^~~~~~~~
src/game/server/gamecontext.cpp:1795:41: error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]
     Converse(ClientID, pMsg->m_pMessage + 3);
                        ~~~~~~~~~~~~~~~~~^~~
In file included from src/game/server/gamecontext.cpp:11:0:
src/game/server/gamecontext.h:267:7: note:   initializing argument 2 of ‘void CGameContext::Converse(int, char*)’
  void Converse(int ClientID, char *pStr);
       ^~~~~~~~
src/game/server/gamecontext.cpp: In member function ‘void CGameContext::Converse(int, char*)’:
src/game/server/gamecontext.cpp:4533:10: error: return-statement with a value, in function returning 'void' [-fpermissive]
   return false;
          ^~~~~
src/game/server/gamecontext.cpp:4545:47: error: ‘Buffer’ was not declared in this scope
   TextIter = FinalMessage.append_at(TextIter, Buffer.buffer());
                                               ^~~~~~
bam: 'objs/game/server/gamecontext_sql.o' error 256
```

Comment (BloodWod) [reply src/game/server/gamecontext.cpp]: look at the commit "Fix: fix bugs"

**Comment by bretonium** [comment]
```
[2020-06-03 01:32:33][chat]: σℓí♡: breton (private): через /w
[2020-06-03 01:32:38][chat]: breton: breton (private): через /c
```
Pay attention that in the second line breton writes to breton

**Pull request closed without merging**